File: RoamNote/interval scheduling/Container With Most Water/code/Container_With_Most_Water.go
package main

func main() {
//go version
import "math"
func maxArea(height []int) int {
    front := 0
    end := len(height)-1

    var maxCompacity float64 = 0

    for front < end {
        longSide := math.Min(float64(height[front]),float64(height[end]))
        shotSide := float64(end-front)
        tmpCompacity := longSide*shotSide
        maxCompacity = math.Max(tmpCompacity,maxCompacity)
        if height[front] >= height[end] {
            end--
        }else{
            front++
        }
    }
    return int(maxCompacity)
}
}

File: RoamNote/interval scheduling/Container With Most Water/code/Container_With_Most_Water.cpp
//c++ version
#include <algrothim>
#include <iostream>
#include <math>

using namespace std;

class Solution {
public:
    int maxArea(vector<int> &height) {
        int left = 0;
        int right = height.size()-1;
        int compacity = 0;

        while (left < right) {
            int min = fmin(height[left], height[right]);
            int length = right - left;
            int tmpCompacity = min * length;

            if (tmpCompacity >= compacity)
                compacity = tmpCompacity;
            if (height[left] >= height[right])
                right--;
            else
                left++;
        }

        return compacity;
    }
};

int main(int argc, char *argv[]) { return 0; }
